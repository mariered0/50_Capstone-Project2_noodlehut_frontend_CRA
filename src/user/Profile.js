import React, { useContext, useState } from "react";
import userContext from "./UserContext";
import { useFormik } from "formik";
import { updateSchema } from "../schemas/updateSchema";
import NoodleHutApi from "../api/api";
import ErrorMessage from "../common/ErrorMessage";
import SuccessMessage from "../common/SuccessMessage";
import { Container, Avatar, Typography, Grid, TextField, Button, Box,
} from "@mui/material";
import AccountCircleIcon from "@mui/icons-material/AccountCircle";

const Profile = () => {
  const { currentUser, setCurrentUser } = useContext(userContext);
  const [submitErrors, setErrors] = useState(null);
  const [saveSuccess, setSaveSuccess] = useState(false);

  const onSubmit = async (values, actions, e) => {
    const data = {
        firstName: values.firstName,
        lastName: values.lastName,
        phone: values.phone,
        email: values.email,
        isAdmin: false
    };

    try {
        const user = await NoodleHutApi.updateCurrentUserProf(currentUser.username, data);
        setCurrentUser(user);
        setErrors(null);
        setSaveSuccess(true);
    }catch (err){
        //Set server-side error to state.
      setErrors(submitErrors => [err]);
    }
  };

  const { handleBlur, errors, touched, handleChange, handleSubmit, values } = useFormik(
    {
        initialValues: {
            username: currentUser.username,
            firstName: currentUser.firstName,
            lastName: currentUser.lastName,
            phone: currentUser.phone,
            email: currentUser.email,
            isAdmin: false
      },
      validationSchema: updateSchema,
      onSubmit,
    }
  );

  return (
    <>
      <Container component="main" maxWidth="xs">
      {saveSuccess && <SuccessMessage content={`Information updated successfully!`}/>}
        <Box
          sx={{
            marginTop: saveSuccess && 2 || 8,
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
          }}
        > 
          <Avatar sx={{ m: 1, bgcolor: "secondary.main" }}>
            <AccountCircleIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Account Information
          </Typography>

          {submitErrors && <ErrorMessage error={submitErrors} />}

          <Box
            component="form"
            noValidate
            onSubmit={handleSubmit}
            sx={{ mt: 3 }}
          >
            <Grid container spacing={2}>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="firstName"
                  variant="outlined"
                  required
                  fullWidth
                  id="firstName"
                  value={values.firstName}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  error={errors.firstName && touched.firstName ? true : false}
                  helperText={touched.firstName ? errors.firstName : ""}
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="lastName"
                  value={values.lastName}
                  name="lastName"
                  autoComplete="lname"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  error={errors.lastName && touched.lastName ? true : false}
                  helperText={touched.lastName ? errors.lastName : ""}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="username"
                  value={values.username}
                  disabled={true}
                  name="username"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  error={errors.username && touched.username ? true : false}
                  helperText={touched.username ? errors.username : ""}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="email"
                  name="email"
                  autoComplete="email"
                  value={values.email}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  error={errors.email && touched.email ? true : false}
                  helperText={touched.email ? errors.email : ""}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="phone"
                  name="phone"
                  autoComplete="phone"
                  value={values.phone}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  error={errors.phone && touched.phone ? true : false}
                  helperText={touched.phone ? errors.phone : ""}
                />
              </Grid>
            </Grid>

            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
            >
              Save
            </Button>
            <Grid container justifyContent="flex-end">
            </Grid>
          </Box>
        </Box>
      </Container>
    </>
  );
};

export default Profile;
