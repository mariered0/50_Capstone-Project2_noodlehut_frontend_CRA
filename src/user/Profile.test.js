import React from "react";
import { render } from '@testing-library/react';
import Profile from "./Profile";
import UserContext from "../user/UserContext";
const { currentUser, setCurrentUser } = require("../cart/_testCommon");


it("renders without crashing", () => {
    render(
        <UserContext.Provider value={{ currentUser, setCurrentUser }}>
            <Profile />
        </UserContext.Provider>
        
    );
});

it("matches snapshot", () => {
	const { asFragment } = render(
        <UserContext.Provider value={{ currentUser, setCurrentUser }}>
            <Profile />
        </UserContext.Provider>
    );
	expect(asFragment()).toMatchSnapshot();
});