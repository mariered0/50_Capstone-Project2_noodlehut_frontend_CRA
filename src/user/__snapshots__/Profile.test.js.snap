// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-binzgt"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccountCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Account Information
      </h1>
      <form
        class="MuiBox-root css-178yklu"
        novalidate=""
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="firstName"
                  name="firstName"
                  required=""
                  type="text"
                  value="Test"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="lname"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="lastName"
                  name="lastName"
                  required=""
                  type="text"
                  value="Test"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id="username"
                  name="username"
                  required=""
                  type="text"
                  value="test"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  required=""
                  type="text"
                  value="test@gmail.com"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="phone"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="phone"
                  name="phone"
                  required=""
                  type="text"
                  value="0000000000"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Save
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container css-16g5rpk-MuiGrid-root"
        />
      </form>
    </div>
  </main>
</DocumentFragment>
`;
